libs/auth/testing/src/inmemory-backend/auth.service.unit.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { TestBed } from '@angular/core/testing';
+import { STATUS } from 'angular-in-memory-web-api';
+
+import { DaffInMemoryBackendAuthService } from './auth.service';
+import { DaffAccountRegistrationFactory } from '../factories/account-registration.factory';
+
+describe('DaffAuthInMemoryBackend | Unit', () => {
+  let authTestingService;
+  let reqInfoStub;
+
+  const registrationFactory = new DaffAccountRegistrationFactory();
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        DaffInMemoryBackendAuthService,
+      ]
+    });
+
+    authTestingService = TestBed.get(DaffInMemoryBackendAuthService);
+
+    reqInfoStub = {
+      utils: {
+        createResponse$: f => f(),
+        getJsonBody: () => ({})
+      }
+    };
+  });
+
+  it('should be created', () => {
+    expect(authTestingService).toBeTruthy();
+  });
+
+  describe('after intializiaton', () => {
+    let result;
+
+    beforeEach(() => {
+      result = authTestingService.createDb();
+    });
+
+    it('should have any empty database', () => {
+      expect(result.auth).toEqual({});
+    });
+  });
+
+  describe('processing a login request', () => {
+    it('should process post requests of the form `/api/auth/login` and return a token with an OK status', () => {
+      reqInfoStub.id = 'login';
+      const result = authTestingService.post(reqInfoStub);
+
+      expect(result.body).toBeDefined();
+      expect(result.status).toEqual(STATUS.OK);
+    });
+  });
+
+  describe('processing a register request', () => {
+    it('should process post requests of the form `/api/auth/register` and return the customer info with a CREATED status', () => {
+      reqInfoStub.id = 'register';
+      reqInfoStub.utils.getJsonBody = () => registrationFactory.create();
```

Comment:
Let's just return an empty success response on creating an account, or potentially, just return something that looks like a `guid` since that's fairly common amongst providers.

Reply:
Can we set the `body` manually, or do we have to rely on this odd internal mechanism?

Reply:
I can set `reqInfoStub.req.body` but that would require that I set `reqInfoStub.utils.getJsonBody` to mimic to the behavior of the real `RequestInfo`. Unless there is a way to generate mocked `RequestInfo`s that I'm not aware of, I have to deal with `getJsonBody` at some point.